bin/cron/applab_datasets/spotify

```diff
@@ -4,100 +4,132 @@ abort 'Script already running' unless only_one_running?(__FILE__)
 
 # This script fetches spotify data from api.spotify and uploads to the shared firebase channel.
 require_relative '../../../deployment'
-require 'net/http'
-require 'csv'
 require 'ostruct'
 require_relative '../../../shared/middleware/helpers/firebase_helper'
 require 'cdo/profanity_filter'
+require 'rest-client'
+
+BASE_URL = 'https://api.spotify.com/v1'
+
+#PLAYLIST IDs
+VIRAL_50_GLOBAL_ID = '37i9dQZEVXbLiRSasKsNU9'
+VIRAL_50_USA_ID = '37i9dQZEVXbKuaTI1Z1Afx'
+TOP_50_GLOBAL_ID = '37i9dQZEVXbMDoHDwVN2tF'
+TOP_50_USA_ID = '37i9dQZEVXbLRQDuF5jeBp'
 
-TOP_200_US_URL = 'https://spotifycharts.com/regional/us/daily/latest/download'
-TOP_200_WORLD_URL = 'https://spotifycharts.com/regional/global/daily/latest/download'
-VIRAL_50_US_URL = 'https://spotifycharts.com/viral/us/daily/latest/download'
-VIRAL_50_WORLD_URL = 'https://spotifycharts.com/viral/global/daily/latest/download'
 PROFANITY_LOCALE = 'en-US'
 
-def get_top200_data(url)
-  records = {}
-  columns = ['id', 'Track Name', 'Artist', 'Position', 'Streams', 'URL']
-  response = Net::HTTP.get_response(URI(url))
-  return unless response.code == '200'
-  id = 1
-  response.body.force_encoding('utf-8')
-  csv = CSV.new(response.body)
-  lines = csv.read
-  lines.shift # first line just contains a note
-  lines.shift # second line is column names
-  regex = generate_profanity_regex(lines)
-  lines.each do |line|
-    next if regex&.match?(get_string_to_check(line)) # skip record if it contains profanity
+CHART_TYPE = {
+  TOP: 'TOP',
+  VIRAL: 'VIRAL',
+}
 
-    record = OpenStruct.new
-    record.id = id
-    record.Position = line[0].to_i
-    record['Track Name'] = line[1]
-    record.Artist = line[2]
-    record.Streams = line[3].to_i
-    record.URL = line[4]
+def get_token
+  client_id = CDO.spotify_api_client_id
+  client_secret = CDO.spotify_api_client_secret
 
-    records[id] = record.to_h.to_json
-    id += 1
+  begin
+    url = "https://accounts.spotify.com/api/token"
+    headers = {
+      'authorization' => 'Basic ' + Base64.strict_encode64(client_id + ':' + client_secret),
+      'content-type' => 'application/x-www-form-urlencoded',
+    }
+    form = {
+      'grant_type' => 'client_credentials'
+    }
+    auth_response = JSON.parse(RestClient.post(url, form, headers))
+    $token = auth_response['access_token']
+  rescue RestClient::Exception => e
+    Honeybadger.notify(
+      e,
+      error_message: "Failed to retrieve OAuth token from Spotify for use in App Lab Datasets.",
+      context: {
+        tenant_id: client_id,
+        url: url
+      }
+    )
   end
```

Comment:
Added logging after seeing a separate ticket to add logging for a different API we use for datasets: https://codedotorg.atlassian.net/browse/STAR-1707